ci-operator/step-registry/hypershift/hostedcluster/create/hostedcluster/hypershift-hostedcluster-create-hostedcluster-commands.sh

```diff
@@ -46,28 +53,51 @@ if [[ "${COMPUTE_NODE_TYPE}" == "" ]]; then
 fi
 
 echo "$(date) Creating HyperShift cluster ${CLUSTER_NAME}"
-/usr/bin/hypershift create cluster aws \
-  ${EXTRA_ARGS} \
-  --name ${CLUSTER_NAME} \
-  --infra-id ${INFRA_ID} \
-  --node-pool-replicas ${HYPERSHIFT_NODE_COUNT} \
-  --instance-type=${COMPUTE_NODE_TYPE} \
-  --base-domain ${DOMAIN} \
-  --region ${HYPERSHIFT_AWS_REGION} \
-  --control-plane-availability-policy ${HYPERSHIFT_CP_AVAILABILITY_POLICY} \
-  --infra-availability-policy ${HYPERSHIFT_INFRA_AVAILABILITY_POLICY} \
-  --pull-secret=/tmp/pull-secret.json \
-  --aws-creds=${AWS_GUEST_INFRA_CREDENTIALS_FILE} \
-  --release-image ${RELEASE_IMAGE} \
-  --control-plane-operator-image=${CONTROLPLANE_OPERATOR_IMAGE:-} \
-  --node-selector "hypershift.openshift.io/control-plane=true" \
-  --additional-tags="expirationDate=$(date -d '4 hours' --iso=minutes --utc)" \
-  --annotations "prow.k8s.io/job=${JOB_NAME}" \
-  --annotations "prow.k8s.io/build-id=${BUILD_ID}" \
-  --annotations resource-request-override.hypershift.openshift.io/kube-apiserver.kube-apiserver=memory=3Gi,cpu=2000m \
-  --annotations hypershift.openshift.io/cleanup-cloud-resources="false" \
-  --additional-tags "prow.k8s.io/job=${JOB_NAME}" \
-  --additional-tags "prow.k8s.io/build-id=${BUILD_ID}"
+if [[ "${PLATFORM}" == "aws" ]]; then
+  /usr/bin/hypershift create cluster aws \
+    ${EXTRA_ARGS} \
+    --name ${CLUSTER_NAME} \
+    --infra-id ${INFRA_ID} \
+    --node-pool-replicas ${HYPERSHIFT_NODE_COUNT} \
+    --instance-type=${COMPUTE_NODE_TYPE} \
+    --base-domain ${DOMAIN} \
+    --region ${HYPERSHIFT_AWS_REGION} \
+    --control-plane-availability-policy ${HYPERSHIFT_CP_AVAILABILITY_POLICY} \
+    --infra-availability-policy ${HYPERSHIFT_INFRA_AVAILABILITY_POLICY} \
+    --pull-secret=/tmp/pull-secret.json \
+    --aws-creds=${AWS_GUEST_INFRA_CREDENTIALS_FILE} \
+    --release-image ${RELEASE_IMAGE} \
+    --control-plane-operator-image=${CONTROLPLANE_OPERATOR_IMAGE:-} \
+    --node-selector "hypershift.openshift.io/control-plane=true" \
+    --additional-tags="expirationDate=$(date -d '4 hours' --iso=minutes --utc)" \
+    --annotations "prow.k8s.io/job=${JOB_NAME}" \
+    --annotations "prow.k8s.io/build-id=${BUILD_ID}" \
+    --annotations resource-request-override.hypershift.openshift.io/kube-apiserver.kube-apiserver=memory=3Gi,cpu=2000m \
+    --annotations hypershift.openshift.io/cleanup-cloud-resources="false" \
+    --additional-tags "prow.k8s.io/job=${JOB_NAME}" \
+    --additional-tags "prow.k8s.io/build-id=${BUILD_ID}" 
+else
```

Comment:
can we handle this like how its handled in destroy?

Reply:
We have added this check already in the early parts of the script,
if the platform is not correct, the script exits before reaching the create command.
https://github.com/openshift/release/pull/43673/files#diff-6f6f9668e008a46382d5523177e1aeeb3f3d09f9c92baee608a2256b713552f6R24
And, so we do not need this check again here.